### ISSUE TITLE
Internal Server Error when appName contains chinese character

### ISSUE THREAD

**Issue by yuntu800** [Microsoft/cordova-plugin-code-push#257]
It was worked well before，but I got internal server error recently。I guess it caused by  chinese character App name, 
```

E:\>code-push deployment history 中文名称AppName Production
[Error]  Internal Server Error    ----this app does not exist, but return error

E:\>code-push deployment history englishAppName Production
[Error]  App "englishAppName" does not exist.

E:\>code-push app rename 中文 testApp
[Error]  Internal Server Error

E:\>code-push -v
2.0.1-beta`
```

**Comment by sergey-akhalkov** [comment]
Hi @yuntu800, thanks for reaching us. Do I get it right if have registered CodePush app that uses Chinese characters in it's name and currently you are not able to get deployment history for your app? If so - we are going to investigate and try to fix it. We'll let you know if any changes.

Currently users can use appName that  contains only alphanumeric characters, dashes, periods, or underscores, otherwise they will get the following error:
```
code-push app add 中文名称AppName android react-native
[Error]  The app name "中文名称AppName" isn't valid. It can only contain alphanumeric characters, dashes, periods, or underscores.
```

**Comment by sergey-akhalkov** [comment]
@yuntu800, could you also please perform `code-push app ls` and share us with its output?

**Comment by yuntu800** [comment]
@sergey-akhalkov 

E:\>code-push app ls
┌───────────────────────────────────────────┬────────────────────
─┐
│ Name                                      │ Deployments         │
├───────────────────────────────────────────┼────────────────────
─┤
│ ba1-bai3-yun2-tu2-android  (八佰云图-android) │ Production, Staging │
├───────────────────────────────────────────┼────────────────────
─┤
│ ba1-bai3-yun2-tu2-IOS  (八佰云图-IOS)         │ Production, Staging │
└───────────────────────────────────────────┴────────────────────
─┘

**Comment by BretJohnson** [comment]
As an update: There are a couple bugs here.
1. We needed to URL encode calls to Mobile Center. That fixes the "Internal Server Error" on app names with non-ASCII characters.
2. There was a problem with app rename code.
I've fixed both bugs. We'll deploy a server update very shortly with the fixes. I'll update this issue after it's deployed.

**Comment by BretJohnson** [comment]
HI. I've updated the CodePush server, with both of the fixes above.
@yuntu800 - Can you please try again and let us know if you still see any issues.

A note on how app names work now in Mobile Center:
In Mobile Center apps have two names, the display name and the link name. Typically they are the same, especially for CodePush users, but if they are different you will see the the link name listed first when doing `code-push app ls`, followed by the display name in parentheses. The link name can only contain alphanumeric characters, dashes, periods, or underscores, while the display name can contain any characters.  In Mobile Center, the link name is used for REST APIs, shared app URLs, and from the Mobile Center CLI--all things that are used in automation scenarios. The display name is shown in the web UI.

We encourage folks to use the link name in the code-push CLI, but using either link name or the display name will actually work there.

So, to sum up, using either `ba1-bai3-yun2-tu2-android` (the link name) or `八佰云图-android` (the display name) should work from the CLI now.  You can also rename the app, if you choose. Use `code-push app rename` to change the link name and the web portal UI to change the display name.

**Comment by yuntu800** [comment]
@BretJohnson CLI works well now, thanks.
